sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/SkipExternalizeTestRecording.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation given to some tests to skip externalizing recording to enable running test validations locally.
+ * <p>This requires the test recordings to be present src/test/resources/session-records folder</p>
+ */
+@Retention(RUNTIME)
+@Target(ElementType.METHOD)
```

Comment:
We should support having this annotation on the entire test class.

Reply:
Would we ever have scenarios for the entire test class to not be externalized?